app/assets/javascripts/arclight/context_navigation.js

```diff
@@ -55,6 +90,182 @@ class ContextNavigation {
+
+    // Update the docs before the item
+    // Retrieves the documents up to and including the "new document"
+    const beforeDocs = newDocs.slice(0, newDocIndex);
+    let renderedBeforeDocs;
+    if (beforeDocs.length > 1) {
+      beforeDocs.forEach(function (parentDoc) {
+        parentDoc.collapse();
+      });
+      renderedBeforeDocs = beforeDocs.map(newDoc => newDoc.render()).join('');
+      const prevParentContainer = this.buildExpandContainer();
+      prevParentContainer.append(renderedBeforeDocs);
+    } else {
+      renderedBeforeDocs = beforeDocs.map(newDoc => newDoc.render()).join('');
+    }
+
+    parentLi.before(renderedBeforeDocs).fadeIn(500);
+
+    let itemDoc = newDocs.slice(newDocIndex, newDocIndex + 1);
+    let renderedItemDoc = itemDoc.map(doc => doc.render()).join('');
+
+    // Update the item
+    const $itemDoc = $(renderedItemDoc);
+    // Update the id, add classes of the classes. Prepend the current children.
+    parentLi.attr('id', $itemDoc.attr('id'));
+    parentLi.addClass($itemDoc.attr('class'));
+
+    parentLi.prepend($itemDoc.children()).fadeIn(500);
+
+    // Update the docs after the item
+    const afterDocs = newDocs.slice(newDocIndex + 1, newDocs.length);
+    const renderedAfterDocs = afterDocs.map(newDoc => newDoc.render()).join('');
+
+    // Insert the documents after the current
+    parentLi.after(renderedAfterDocs).fadeIn(500);
+  }
+
+  /**
+   * Update the ancestors for <li> elements
+   * @param {jQuery} $li - the <li> element for the current, highlighted
+   *   Document in the <ul> context list of collections, components, and
+   *   containers
+   */
+  updateListSiblings($li) {
+    const prevSiblings = $li.prevAll('.al-collection-context');
+    /**
+     * @todo This should be deduplicated and refactored - perhaps a Class
+     *   derived from ExpandButton?
+     */
+    if (prevSiblings.length > 1) {
+      const hiddenNextSiblings = prevSiblings.slice(0, -1);
+      hiddenNextSiblings.toggleClass('collapsed');
+
+      // This all needs to be refactored
+      const $button = $('<button class="my-3 btn btn-secondary btn-sm">Expand</button>');
+      $button.handleClick = event => {
+        const highlighted = $(this.parentLi).siblings('.al-hierarchy-highlight');
+        let targeted = highlighted.prevAll('.al-collection-context');
+        targeted = targeted.slice(0, -1);
+        targeted.toggleClass('collapsed');
+        const collapsed = targeted.hasClass('collapsed');
+        const updatedText = collapsed ? 'Expand' : 'Collapse';
+        const $target = $(event.target);
+        $target.text(updatedText);
+      };
+      $button.handleClick = $button.handleClick.bind($button);
+      $button.click($button.handleClick);
+
+      const lastHiddenNextSibling = hiddenNextSiblings[hiddenNextSiblings.length - 1];
+      $button.insertAfter(lastHiddenNextSibling);
+    }
+
+    const nextSiblings = $li.nextAll('.al-collection-context');
```

Comment:
I'm not certain if this was actually desired (given what was in the wireframes).